### ISSUE TITLE
[Feature Request] [Longterm] Replace const-ref std::string arg with std::string_view

### ISSUE THREAD

**Issue by elad335** [RPCS3/rpcs3#10110]
The difference between the two is that reference references an *object* of type std::string which is defined to contain a unique buffer for its characters, while std::string_view describes a section of existing memory containing the characters. The previous arguments decleration style makes passing substring and static C-string to a function include an additional and unneccessary step of allocating new buffer for and copying the charavters to it, std::string_view argument does not have this flaw as it is capable of describing these sections of memory without copying them.

**Comment by elad335** [comment]
This is true for even `static constexpr std::string` in C++20 when an address is required such as when calling c_str() (compiler cannot optimize it). So string_view is recommended for static strings as well.

**Comment by Nekotekina** [comment]
For constexpr strings plain old char array can do its job. It's usually not a problem to convert it to string_view on the fly. It feels however we may be missing some kind of `nts_view` to preserve a null-terminator.

**Comment by elad335** [comment]
Most have been replaced by now.
